ocaml/typing/layouts.ml

```diff
@@ -744,7 +752,12 @@ module Layout = struct
       | Extensible_variant -> fprintf ppf "it's an extensible variant"
       | Primitive id ->
         fprintf ppf "it equals the primitive value type %s" (Ident.name id)
-      | Type_argument -> fprintf ppf "a type argument defaults to layout value"
+      | Type_argument parent_path ->
+        fprintf ppf "the type argument of %a has layout value"
+          !printtyp_path parent_path
+      | Imported_type_argument parent_path ->
```

Comment:
Interesting. My simplification above suggests we should recombine `Type_argument` and `Imported_type_argument`. I can't think of a good reason why not, so let's just do that.